lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -324,16 +327,46 @@ def _on_update_status(self, event: UpdateStatusEvent):
         # handle when/if certificates are expired
         self._check_certs_expiration(event)
 
-    def _on_get_admin_secrets_action(self, event: ActionEvent):
+    def _on_set_password_action(self, event: ActionEvent):
+        """Set new admin password from user input or generate if not passed."""
+        if not self.unit.is_leader():
+            event.fail("The action can be run only on leader unit.")
+            return
+
+        user_name = event.params.get("username")
+        if event.params.get("username") != "admin":
+            event.fail("Only the 'admin' username is allowed for this action.")
+            return
+
+        password = event.params.get("password") or generate_password()
+        try:
+            self._put_admin_user(password)
+            password = self.secrets.get(Scope.APP, "admin_password")
+            event.set_results({f"{user_name}-password": password})
```

Comment:
**suggestion**  Uhm, maybe I should remove the "-password". I suppose it is already fairly clear in the UX

Reply:
I also agree, but it seems that all the other charms implement it this way, so i'm following their UX for consistency

Reply:
yes, indeed!